internal/operands/common-templates/reconcile.go

```diff
@@ -239,6 +245,10 @@ func reconcileTemplatesFuncs(templatesBundle []templatev1.Template) []common.Rec
 					newTemplate := newRes.(*templatev1.Template)
 					foundTemplate := foundRes.(*templatev1.Template)
 
+					if foundTemplate.GetLabels()[common.AppKubernetesVersionLabel] == newTemplate.GetLabels()[common.AppKubernetesVersionLabel] {
```

Comment:
At this point in the code these labels are guaranteed to be the same, so this condition is always `true`.
Look here: https://github.com/kubevirt/ssp-operator/blob/6cb14a15b4d1169fddd412193da0aa4ff58d83d0/internal/common/resource.go#L185-L191

The `UpdateLabels()` function is called before `r.updateFunc()`.